patches/api/0449-add-number-format-api.patch

```diff
@@ -0,0 +1,185 @@
++        return new FixedFormatImpl(component);
++    }
++}
+diff --git a/src/main/java/io/papermc/paper/scoreboard/numbers/StyledFormat.java b/src/main/java/io/papermc/paper/scoreboard/numbers/StyledFormat.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..3c644e150bed9df74e9168b2fcbbaab58cb6a654
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/scoreboard/numbers/StyledFormat.java
+@@ -0,0 +1,23 @@
++package io.papermc.paper.scoreboard.numbers;
++
++import net.kyori.adventure.text.format.NamedTextColor;
++import net.kyori.adventure.text.format.Style;
++
++/*
++ * A scoreboard number format that applies a custom formatting to the score number.
++ */
++public interface StyledFormat extends NumberFormat {
++
++    StyledFormat NO_STYLE = NumberFormat.styled(Style.empty());
++
++    StyledFormat SIDEBAR_DEFAULT = NumberFormat.styled(Style.empty().color(NamedTextColor.RED));
++
++    StyledFormat PLAYER_LIST_DEFAULT = NumberFormat.styled(Style.empty().color(NamedTextColor.YELLOW));
++
++    /*
++     * The style that is being applied to the number in the score
++     * @return the style to apply
++     */
++    Style style();
++
++}
+diff --git a/src/main/java/io/papermc/paper/scoreboard/numbers/StyledFormatImpl.java b/src/main/java/io/papermc/paper/scoreboard/numbers/StyledFormatImpl.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..2489f63707bce74ecd8045948de73d529c06a446
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/scoreboard/numbers/StyledFormatImpl.java
+@@ -0,0 +1,5 @@
++package io.papermc.paper.scoreboard.numbers;
++
++import net.kyori.adventure.text.format.Style;
++
++record StyledFormatImpl(Style style) implements StyledFormat {}
+diff --git a/src/main/java/org/bukkit/scoreboard/Objective.java b/src/main/java/org/bukkit/scoreboard/Objective.java
+index a193ffabb05160b462dee1ba8f687fdbc84405b6..703b7788fbed7aff7f8274f40df3d4b6bfa30804 100644
+--- a/src/main/java/org/bukkit/scoreboard/Objective.java
++++ b/src/main/java/org/bukkit/scoreboard/Objective.java
+@@ -175,4 +175,20 @@ public interface Objective {
+      */
+     @NotNull Score getScoreFor(@NotNull org.bukkit.entity.Entity entity) throws IllegalArgumentException, IllegalStateException;
+     // Paper end - improve scoreboard entries
++
++    //Paper start - number format
++    /*
++     * Gets the number format for this objective's entries (scores)
++     * @return this objective's number format
++     * @throws IllegalStateException if this objective has been unregistered
++     */
++    @Nullable io.papermc.paper.scoreboard.numbers.NumberFormat getNumberFormat();
++
++    /*
++     * Sets the number format for this objective's entries (scores)
++     * @param format the number format to set, pass null to reset format to default
++     * @throws IllegalStateException if this objective has been unregistered
++     */
++    void setNumberFormat(@Nullable io.papermc.paper.scoreboard.numbers.NumberFormat format);
```

Comment:
I think we are better off sticking to fluent naming here.
While spigot has not released their API yet, potentially clashing with them seems like a thing we might want to avoid.

Reply:
> I think we are better off sticking to fluent naming here. While spigot has not released their API yet, potentially clashing with them seems like a thing we might want to avoid.

With fluent naming, you mean numberFormat() as a getter and numberFormat(param) as a setter, like in the adventure APIs?

Reply:
Jup, that is fluent naming.